flent/runners.py

```diff
@@ -845,6 +846,95 @@ def find_binary(cls, host, interval, length, ip_version, local_bind=None,
+
+    time_re = re.compile(r"^Time: (?P<timestamp>\d+\.\d+)", re.MULTILINE)
+    cwnd_re = re.compile(r"cwnd:(?P<cwnd>\d+)", re.MULTILINE)
+    rtt_re = re.compile(r"rtt:(?P<rtt>\d+.\d+/\d+.\d+)", re.MULTILINE)
+    pid_re = re.compile(r"pid=(?P<pid>\d+)", re.MULTILINE)
+    ports_ipv4_re = re.compile(r"" + ip_v4_addr_sub_re +
+                               ":(?P<src_p>\d+)\s+?" + ip_v4_addr_sub_re + ":(?P<dst_p>\d+)")
+    ports_ipv6_re = re.compile(r"" + "(?P<src_p>" + ip_v6_addr_sub_re + ")" + "\s+" +
+                               "(?P<dst_p>" + ip_v6_addr_sub_re + ")", re.IGNORECASE)
+
+    src_p = []
+    dst_p = []
+    pid = []
+
+    def parse_part(self, part):
+        timestamp = self.time_re.search(part)
+        cwnd = self.cwnd_re.search(part)
+        rtt = self.rtt_re.search(part)
+        pid = self.pid_re.search(part)
+
+        f_ports = self.ports_ipv4_re.search(part)
+        if None is f_ports:
+            f_ports = self.ports_ipv6_re.search(part)
+
+        if None in [timestamp, cwnd, rtt, pid, f_ports]:
+            raise ParseException()
+
+        timestamp = float(timestamp.group('timestamp'))
+        cwnd = int(cwnd.group('cwnd'))
+        rtt, rtt_var = [float(x) for x in rtt.group('rtt').split('/')]
+        pid = int(pid.group('pid'))
+        self.src_p.append(f_ports.group('src_p').split(":")[-1])
+        self.dst_p.append(f_ports.group('dst_p').split(":")[-1])
+
+        val_res = {'cwnd': cwnd, 'rtt': rtt, 'rtt_var': rtt_var}
+
+        return timestamp, val_res, pid
+
+    def parse(self, output):
+        results = {}
+        parts = output.split("\n---\n")
+        for part in parts:
+            try:
+                t, res_dict, pid = self.parse_part(part)
+                for k, v in res_dict.items():
+                    if k not in results:
+                        results[k] = [[t, v]]
+                    else:
+                        results[k].append([t, v])
+
+                self.pid.append(pid)
+                raw_values = copy.deepcopy(results)
+                raw_values['t'] = t
+                self.raw_values.append(raw_values)
+            except ParseException:
+                continue
+
+        return results
+
+    @classmethod
+    def find_binary(self, host, interval, length, target, port):
+        script = os.path.join(DATA_DIR, 'scripts', 'ss_iterate.sh')
+        if not os.path.exists(script):
+            raise RuntimeError("Cannot find ss_iterate.sh.")
+
+        bash = util.which('bash')
+        if not bash:
+            raise RuntimeError("Socket stats requires a Bash shell.")
+
+        return "{bash} {script} -I {interval:.2f} " \
+            "-c {count:.0f} -H {host} -t {target}".format(
+                bash=bash,
+                script=script,
+                interval=interval,
+                count=length // interval + 1,
+                host=host,
+                target=target,
```

Comment:
And restricting it to IP only? 

Haaa? ss really requires that on IPv6 address! Hm ... hopefully, I've already complained about the current state of ss in your presence.

Well, either or, escaping or [] is necessary as I understand it from file:///usr/share/doc/iproute2-doc/ss.html. Not both.

Mnor one ...